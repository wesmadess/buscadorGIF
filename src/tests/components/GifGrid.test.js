import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />',()=>{
    
    const category = 'Onr Punch';
    
    test('Debe de mostrarse correctamente',()=>{
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true       
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })    

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs',()=>{
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/imagen/uno.jpg',
            title: 'titulo'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false       
        });        

        const wrapper = shallow(<GifGrid category={category} />);        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })        

})