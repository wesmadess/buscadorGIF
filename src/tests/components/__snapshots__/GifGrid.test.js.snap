// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Onr Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="titulo"
      url="https://localhost/imagen/uno.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Onr Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
